lib/Bio/EnsEMBL/DataCheck/Checks/CheckIDRangeMLSS.pm

```diff
@@ -0,0 +1,132 @@
+    SELECT genomic_align_id, method_link_species_set_id
+    FROM genomic_align
+    WHERE genomic_align_id < (method_link_species_set_id * 10000000000) 
+      OR genomic_align_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_1, $desc_1);
+
+
+  my $desc_2 = "genomic_align_block.genomic_align_block_id all within correct MLSS range";
+  my $sql_2 = q/
+    SELECT genomic_align_block_id, method_link_species_set_id
+    FROM genomic_align_block 
+    WHERE genomic_align_block_id < (method_link_species_set_id * 10000000000) 
+      OR genomic_align_block_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_2, $desc_2);
+
+
+  my $desc_3 = "genomic_align_block.group_id all within correct MLSS range";
+  my $sql_3 = q/
+    SELECT group_id, method_link_species_set_id
+    FROM genomic_align_block 
+    WHERE group_id < (method_link_species_set_id * 10000000000) 
+      OR group_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_3, $desc_3);
+
+
+  my $desc_4 = "genomic_align_tree.node_id all within correct MLSS range";
+  my $sql_4 = q/
+    SELECT DISTINCT(t.node_id), g.method_link_species_set_id 
+    FROM genomic_align_tree t 
+    JOIN genomic_align g 
+      ON g.node_id=t.node_id
+    WHERE t.node_id < (g.method_link_species_set_id * 10000000000) 
+      OR t.node_id > ((g.method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_4, $desc_4);
+
+
+  my $desc_5 = "constrained_element.constrained_element_id all within correct MLSS range";
+  my $sql_5 = q/
+    SELECT constrained_element_id, method_link_species_set_id
+    FROM constrained_element 
+    WHERE constrained_element_id < (method_link_species_set_id * 10000000000) 
+      OR constrained_element_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_5, $desc_5);
+
+
+  my $desc_6 = "dnafrag.dnafrag_id for ancestral frags all within correct MLSS range";
+  my $sql_6 = q/
+    SELECT d.dnafrag_id, g.method_link_species_set_id 
+    FROM dnafrag d 
+    JOIN genomic_align g 
+      ON g.dnafrag_id=d.dnafrag_id
+    WHERE d.name LIKE 'Ancestor%'
+      AND (
+        d.dnafrag_id < (g.method_link_species_set_id * 10000000000) 
+        OR d.dnafrag_id > ((g.method_link_species_set_id + 1) * 10000000000)
+      )
+  /;
+  is_rows_zero($self->dba, $sql_6, $desc_6);
+
+
+  my $desc_7 = "conservation_score.genomic_align_block_id all within correct MLSS range";
+  my $sql_7 = q/
+    SELECT cs.genomic_align_block_id, g.method_link_species_set_id
+    FROM conservation_score cs 
+    JOIN genomic_align_block g 
+      ON g.genomic_align_block_id=cs.genomic_align_block_id
+    WHERE g.genomic_align_block_id < (g.method_link_species_set_id * 10000000000) 
+      OR g.genomic_align_block_id > ((g.method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_7, $desc_7);
```

Comment:
`CheckConservationScore.pm` (critical) already checks that all `conservation_score.genomic_align_block_id` match one `genomic_align_block.genomic_align_block_id`, and you have previously checked that `genomic_align_block.genomic_align_block_id` are within the correct range, so I think this check can be skipped:
```suggestion
```

Reply:
I thought when we discussed this midweek it was decided it made sense to report all tables where changes would be required? Yes, the other check would flag a problem, but this DC wouldn't make it clear you need to ensure consistency with `conservation_score` was my understanding?

Reply:
Sorry if there was a misunderstanding, what I meant is that we want to report all tables affected, but if something is wrong in `conservation_score` and `CheckConservationScore.pm` has passed, then the source of the issue is in `genomic_align_block`. Thus, following the database schema should lead the person to fix it here as well. Furthermore, if you would prefer to leave this, then you need to add a check for `genomic_align.genomic_align_block_id` as well. I will leave it up to you which one of the two options you prefer :)

Reply:
Hmm I think prefer to check all possible cases for consistency so I will add the extra test.